12-http/example3/main.go

```diff
@@ -18,20 +21,46 @@ type user struct {
 	Phone string
 }
 
-// users is a slice of users.
-var users []user
+var (
+	// mu protects users.
+	mu sync.RWMutex
+	// initialize users, so that the output JSON won't be null
+	// prior to adding users.
+	users = make([]user, 0)
+)
```

Comment:
That's idiomatic Go style. _Always_ put a lock in a block ahead of the data it protects. If the lock is part of a struct type declaration, put the lock field immediately before the field(s) it protects, and if there are other fields not protected by the mutex, put a blank line between the protected fields and the other fields.